node-graph/gcore/src/vector/vector_data/indexed.rs

```diff
@@ -0,0 +1,90 @@
+use super::{PointId, SegmentId, VectorData};
+use glam::DVec2;
+use petgraph::graph::{EdgeIndex, NodeIndex, UnGraph};
+use rustc_hash::FxHashMap;
+
+/// All the fixed fields of a point from the point domain.
+pub struct Point {
+	pub id: PointId,
+	pub position: DVec2,
+}
+
+/// Useful indexes to speed up various operations on `VectorData`.
+///
+/// Important: It is the user's responsibility to ensure the indexes remain valid after mutations to the data.
+pub struct VectorDataIndex {
+	/// Points and segments form a graph. Store it here in a form amenable to graph algorithms.
+	///
+	/// Currently, segment data is not stored as it is not used, but it could easily be added.
+	pub(crate) point_graph: UnGraph<Point, ()>,
+	pub(crate) segment_to_edge: FxHashMap<SegmentId, EdgeIndex>,
+	/// Get the offset from the point ID.
+	pub(crate) point_to_offset: FxHashMap<PointId, usize>,
+	// TODO: faces
```

Comment:
Is this still needed or is this comment a leftover?

Reply:
This is leftover.